rest-api/model/database.py

```diff
@@ -67,6 +71,30 @@ def session(self):
     finally:
       sess.close()
 
+  # TODO: at the time of writing, a PR went in to backoff that adds a `max_time` parameter.  This
+  # will be part of backoff 1.5 (we are on backoff 1.4).  When it releases, use it to prevent
+  # ludicriously long retry periods.
+  @backoff.on_exception(backoff.expo,
+                        DBAPIError,
+                        max_tries=RETRY_CONNECTION_LIMIT,
+                        giveup=lambda err: not err.connection_invalidated)
+  def autoretry(self, func):
+    """Runs a function of the db session and attempts to commit.  If we encounter a dropped
+    connection, we retry the operation.  The retries are spaced out using exponential backoff with
+    full jitter.  All other errors are propagated.
+    """
+    session = self.make_session()
```

Comment:
can we not just do:

with self.make_session() as session:
  return func(session)

like we do everywhere else? (This handles rollback, etc. for you due to:

https://github.com/all-of-us/raw-data-repository/blob/618506460889d60f4c4c23902bc3b5130160e8a8/rest-api/model/database.py#L59

Reply:
We actually can, since `self.session()` always re-raises whatever exc it hits.  Nice catch!